subsys/bluetooth/host/monitor.c

```diff
@@ -74,18 +64,99 @@ static struct {
 	atomic_t other;
 } drops;
 
+static void drop_add(uint16_t opcode)
+{
+	switch (opcode) {
+	case BT_MONITOR_COMMAND_PKT:
+		atomic_inc(&drops.cmd);
+		break;
+	case BT_MONITOR_EVENT_PKT:
+		atomic_inc(&drops.evt);
+		break;
+	case BT_MONITOR_ACL_TX_PKT:
+		atomic_inc(&drops.acl_tx);
+		break;
+	case BT_MONITOR_ACL_RX_PKT:
+		atomic_inc(&drops.acl_rx);
+		break;
+#if defined(CONFIG_BT_BREDR)
+	case BT_MONITOR_SCO_TX_PKT:
+		atomic_inc(&drops.sco_tx);
+		break;
+	case BT_MONITOR_SCO_RX_PKT:
+		atomic_inc(&drops.sco_rx);
+		break;
+#endif
+	default:
+		atomic_inc(&drops.other);
+		break;
+	}
+}
+
+#if defined(CONFIG_BT_DEBUG_MONITOR_RTT)
+#include <SEGGER_RTT.h>
+
+#define RTT_BUFFER_NAME CONFIG_BT_DEBUG_MONITOR_RTT_BUFFER_NAME
+#define RTT_BUF_SIZE CONFIG_BT_DEBUG_MONITOR_RTT_BUFFER_SIZE
+
+static void monitor_send(const void *data, size_t len)
+{
+	static uint8_t rtt_buf[RTT_BUF_SIZE];
+	static size_t rtt_buf_offset;
+	struct bt_monitor_hdr *hdr;
+	unsigned int cnt = 0;
+	bool drop;
+
+	/* Drop any packet which cannot fit the buffer */
+	drop = rtt_buf_offset + len > sizeof(rtt_buf);
+	if (!drop) {
+		(void)memcpy(rtt_buf + rtt_buf_offset, data, len);
+	}
+
+	rtt_buf_offset += len;
```

Comment:
Do we need to worry about overflow when adding a `size_t` to another `size_t` multiple times? 

Given that the `rtt_buf_offset` should never go above `sizeof(hdr->data_len) + hdr->data_len`, perhaps we should simply check if `len` or `rtt_buf_offset + len` > sizeof(hdr->data_len) + hdr->data_len`.

Reply:
I would not worry about that. If this happens the there is a problem somewhere else.

`rtt_buf_offset` has to be incremented anyway, because the check `rtt_buf_offset < sizeof(hdr->data_len) + hdr->data_len` checks if data is complete. If not it returns and waits for complete packet

Reply:
> If this happens the there is a problem somewhere else.
Yes, but is that being checked somewhere else? (Not from I can tell), but I guess that can be considered outside the scope of this PR, but I generally just like to do defensive programming to avoid any issues happening silently in the future.